core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -160,10 +167,13 @@ public synchronized void connect() throws IOException, ConnectionFailedException
     disconnect();
     Preconditions.checkState(!mClosed, "Client is closed, will not try to connect.");
 
+    int socketTimeoutMs = Configuration.getInt(
+        PropertyKey.SECURITY_AUTHENTICATION_SOCKET_TIMEOUT_MS);
     int maxConnectsTry = Configuration.getInt(PropertyKey.MASTER_RETRY);
     final int BASE_SLEEP_MS = 50;
     RetryPolicy retry =
         new ExponentialBackoffRetry(BASE_SLEEP_MS, Constants.SECOND_MS, maxConnectsTry);
+    ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Each client will create its own thread pool? Also, we should make the thread daemon threads and shutdown the executor when possible.